src/Microsoft.ML.Data/Transforms/ConversionsExtensionsCatalog.cs

```diff
@@ -386,8 +416,15 @@ public static KeyToValueMappingEstimator MapKeyToValue(this TransformsCatalog.Co
             env.CheckValue(columns, nameof(columns));
             var keys = keyValuePairs.Select(pair => pair.Key);
             var values = keyValuePairs.Select(pair => pair.Value);
-            return new ValueMappingEstimator<TInputType, TOutputType>(env, keys, values,
-                    InputOutputColumnPair.ConvertToValueTuples(columns));
+
+            var lookupMap = DataViewHelper.CreateDataView(catalog.GetEnvironment(), keys, values,
+                ValueMappingTransformer.DefaultKeyColumnName,
+                ValueMappingTransformer.DefaultValueColumnName);
+
+            return new ValueMappingEstimator<TInputType, TOutputType>(catalog.GetEnvironment(), lookupMap,
+                lookupMap.Schema[ValueMappingTransformer.DefaultKeyColumnName],
+                lookupMap.Schema[ValueMappingTransformer.DefaultValueColumnName],
+                InputOutputColumnPair.ConvertToValueTuples(columns));
```

Comment:
there is a bit of repetition here, does it make sense to make it part of the ctor?

Reply:
I feel columns must go with the associated `IDataView`. There should be no column if its `IDataView` hasn't been created.